File: src/FoodOrderSystem.Persistence.MongoDB/CartInfoModel.cs
using System;
using System.Collections.Generic;

namespace FoodOrderSystem.Persistence.MongoDB
{
    public class CartInfoModel
    {
        public string OrderType { get; set; }
        public Guid RestaurantId { get; set; }
        public string RestaurantInfo { get; set; }
        public string RestaurantEmail { get; set; }
        public string RestaurantPhone { get; set; }
        public List<OrderedDishInfoModel> OrderedDishes { get; set; }
    }
}

File: src/Gastromio.App/ClientApp/src/app/order/models/cart-info.model.ts
import {OrderedDishInfoModel} from './ordered-dish-info.model';

export class CartInfoModel {

  constructor(init?: Partial<CartInfoModel>) {
    if (init) {
      Object.assign(this, init);
    }
    this.orderedDishes = this.orderedDishes?.map(dish => new OrderedDishInfoModel(dish));
  }

  public orderType: number;

  public restaurantId: string;

  public restaurantInfo: string;

  public orderedDishes: OrderedDishInfoModel[];

  public clone(): CartInfoModel {
    return new CartInfoModel({
      orderType: this.orderType,
      restaurantId: this.restaurantId,
      restaurantInfo: this.restaurantInfo,
      orderedDishes: this.orderedDishes?.map(dish => dish?.clone())
    });
  }

}
